packages/contracts/deploy/L1/999-TransferOwnerships.ts

```diff
@@ -0,0 +1,79 @@
+import assert from 'assert'
+
+import '@kroma-network/hardhat-deploy-config'
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+import { getDeploymentAddress } from '../../src/deploy-utils'
+
+const deployFn: DeployFunction = async (hre) => {
+  const timelockProxyAddress = await getDeploymentAddress(hre, 'TimeLockProxy')
+
+  await transferProxyAdminOwnership(hre, timelockProxyAddress)
+  await transferSystemConfigOwnership(hre, timelockProxyAddress)
+}
+
+const transferProxyAdminOwnership = async (
+  hre: HardhatRuntimeEnvironment,
+  newOwner: string
+) => {
+  const proxyAdminAddress = await getDeploymentAddress(hre, 'ProxyAdmin')
+  let proxyAdmin = await hre.ethers.getContractAt(
+    'ProxyAdmin',
+    proxyAdminAddress
+  )
+  const currentProxyAdminOwner = await proxyAdmin.owner()
+  if (currentProxyAdminOwner === newOwner) {
+    console.log(
+      'skip the ProxyAdmin owner transfer process because the owner has already been transferred.'
+    )
+    return
+  }
+
+  proxyAdmin = proxyAdmin.connect(
+    hre.ethers.provider.getSigner(currentProxyAdminOwner)
```

Comment:
It is just a question, how does this line get signer information of `currentProxyAdminOwner`? I mean, where from? I think it needs private key or other signer information of `currentProxyAdminOwner`.

Reply:
you're right. It need to add the private key of the proxy admin owner to the hardhat config.